lti_consumer/lti_xblock.py

```diff
@@ -678,10 +683,23 @@ def _get_statici18n_js_url(loader):  # pragma: no cover
                 return text_js.format(lang_code=code)
         return None
 
+    def _get_statici18n_js_url(self, loader):
+        """
+        Return the JavaScript translation file provided by the XBlockI18NService.
+        """
+        if url_getter_func := getattr(self.i18n_service, 'get_javascript_i18n_catalog_url', None):
+            if javascript_url := url_getter_func(self):
+                return javascript_url
+
+        if deprecated_url := self._get_deprecated_i18n_js_url(loader):
+            return self.runtime.local_resource_url(self, deprecated_url)
+
+        return None
+
     def validate_field_data(self, validation, data):
```

Comment:
@ahmadabuwardeh thanks again!

Please revert all refactoring changes to this pull request. We only need the `_get_statici18n_js_url` and `_get_deprecated_i18n_js_url`.